plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/AIEngineRegistry.java

```diff
@@ -116,10 +117,19 @@ public List<EngineDescriptor> getCompletionEngines() {
         return list;
     }
 
+    public Optional<EngineDescriptor> getDefaultCompletionEngineDescriptor() {
+        return getCompletionEngines().stream().filter(EngineDescriptor::isDefault).findFirst();
+    }
+
     public DAICompletionEngine<?> getCompletionEngine(String id) throws DBException {
         EngineDescriptor descriptor = getEngineDescriptor(id);
         if (descriptor == null) {
-            throw new DBException("AI engine '" + id + "' not found");
+            log.warn("Active engine is not present in the configuration, switching to default active engine");
+            Optional<EngineDescriptor> defaultCompletionEngineDescriptor = getDefaultCompletionEngineDescriptor();
+            if (defaultCompletionEngineDescriptor.isEmpty()) {
+                throw new DBException("AI engine '" + id + "' not found");
+            }
+            descriptor = defaultCompletionEngineDescriptor.get();
```

Comment:
I would write it differently to avoid nested mess.

```suggestion
            log.warn("Can't find engine '" + id + "', using the default engine");
            descriptor = getDefaultCompletionEngineDescriptor(); // Assuming you don't use optionals anymore
        }
        if (descriptor == null) {
            throw new DBException("AI engine '" + id + "' not found");
```